src/App.js

```diff
@@ -1,61 +1,149 @@
 import './styles/reset.css'
 import './styles/index.css'
-
 import initialStoreItems from './store-items'
+import { Fragment } from 'react'
+import { useState } from 'react'
 
-/*
-Here's what a store item should look like
-{
-  id: '001-beetroot',
-  name: 'beetroot',
-  price: 0.35
-}
+function App() {
+  const [cart, setCart] = useState([])
+  const [currentFilter, setCurrentFilter] = useState('show-all')
+
+  const filterFruits = initialStoreItems.filter(
+    storeItem => storeItem.type === 'fruit'
+  )
+  const filterVegetables = initialStoreItems.filter(
+    storeItem => storeItem.type === 'veg'
+  )
+
+  let filteredItems = initialStoreItems
+
+  if (currentFilter === 'fruit') {
+    filteredItems = filterFruits
+  } else if (currentFilter === 'veg') {
+    filteredItems = filterVegetables
+  }
+
+  const addItemToCart = item => {
+    if (findCartItem(item)) {
+      alert(`${item.name} is already in the cart`)
+      return
+    }
+    item.quantity = 1
+    setCart([...cart, item])
+  }
 
-What should a cart item look like? 🤔
-*/
+  const incrementCartItemQuantity = item => {
+    if (findCartItem(item)) {
+      item.quantity += 1
+    }
+    setCart([...cart])
+  }
 
-console.log(initialStoreItems)
+  const decrementCartItemQuantity = item => {
+    if (findCartItem(item)) {
+      if (item.quantity > 1) {
+        item.quantity -= 1
+      } else removeItemFromCart(item)
+    }
+    setCart([...cart])
+  }
 
-export default function App() {
-  // Setup state here...
+  const removeItemFromCart = cartItem => {
+    cart.splice(cart.indexOf(cartItem), 1)
+  }
+
+  const findCartItem = cartItem => {
+    return cart.find(item => item === cartItem)
+  }
+
+  const priceItemsInCart = cartArray => {
+    const shoppingArray = cartArray.map(
+      cartItem => cartItem.price * cartItem.quantity
+    )
+    const totalPrice = 0 + shoppingArray.reduce((a, b) => a + b, 0)
+    return totalPrice.toFixed(2)
+  }
+
+  const renderCartItem = cartArray => {
```

Comment:
I would possibly make this its own component?

Reply:
Yes, I agree with you. I left this for the end because I did not want to get confused and then I ran out of time. I will be splitting it in its own component this afternoon.